### ISSUE TITLE
Emit inflation events

### ISSUE THREAD

**Issue by sug0** [anoma/namada#3252]
In `FinalizeBlock`, we should emit balance change events related to PoS and PGF inflation.

**Comment by brentstone** [comment]
@opsecx what do you mean lack of transaction?

**Comment by opsecx** [comment]
> @opsecx what do you mean lack of transaction?

My understanding (correct me if wrong) is that pgf emissions when proposals are executed basically just "happens", ie there is no transaction with a tx hash we can go back to, as in most other transactions happening on chain. (any type of transaction really). Seem to remember we discussed it a few months ago where it was clarified this was the case, I could be remembering it wrong, haven't been able to find any such event in my indexer on a surface look.

**Comment by sug0** [comment]
> > [@opsecx](https://github.com/opsecx) what do you mean lack of transaction?
> 
> My understanding (correct me if wrong) is that pgf emissions when proposals are executed basically just "happens", ie there is no transaction with a tx hash we can go back to, as in most other transactions happening on chain. (any type of transaction really). Seem to remember we discussed it a few months ago where it was clarified this was the case, I could be remembering it wrong, haven't been able to find any such event in my indexer on a surface look.

that's right, it's a protocol action, which currently doesn't have that much user feedback, hence this issue

**Comment by sug0** [comment]
Like I said, there is no transaction associated with these changes to the global state. These are protocol actions triggered once per epoch, which can happen with empty blocks (i.e. blocks that have no transactions). The state modifications are somewhat opaque right now, unless you directly compare the state before and after an epoch change. Events emitted upon inflation events would help to identify what state was actually changed.
